src/Avalonia.Base/Rendering/Composition/Expressions/Expression.cs

```diff
@@ -27,6 +28,7 @@ public virtual void CollectReferences(HashSet<(string parameter, string property
         protected abstract string Print();
         public override string ToString() => Print();
 
+        [UnconditionalSuppressMessage("Trimming", "IL3050", Justification = TrimmingMessages.DesignTimeSupressWarningMessage)]
```

Comment:
Nit: should we change the code below from `GetMember(...)[0].GetCustomAttribute()` to `GetMember(...).FirstOrDefault()?.GetCustomAttribute()` for safety? I'm aware it shouldn't normally be called, but since it's `ToString()`, it doesn't hurt making it trimming-proof.

Reply:
Ideally, we should simply remove PrettyPrintStringAttribute usage and just put a switch-tree there. Because why not, it's not a public API that might need to have customizable "pretty print" members.

Reply:
Not in this PR though, I would say.

Reply:
Agreed to both points.